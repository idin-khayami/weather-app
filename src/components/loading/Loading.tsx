import { ComponentBaseProps } from '@/types/global.types';

interface LoadingProps extends ComponentBaseProps {
  isLoading?: boolean;
}

export const Loading = ({ isLoading = true, dataTestId }: LoadingProps) => {
  if (!isLoading) return null;

  return <p data-testid={dataTestId}>Loading ...</p>;
};
