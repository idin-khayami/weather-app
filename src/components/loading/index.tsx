import { Loading } from '@/components/loading/Loading';

export { Loading };
