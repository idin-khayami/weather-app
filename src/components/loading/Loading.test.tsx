import { render, screen } from '@testing-library/react';

import { Loading } from '@/components/loading/Loading';

describe('Component: loading component', () => {
  it('should match the snapshot with loading state', () => {
    render(<Loading dataTestId="testingLoading" />);

    const loading = screen.getByTestId('testingLoading');

    expect(loading).toBeInTheDocument();
    expect(loading).toMatchSnapshot();
  });

  it('should match the snapshot without loading state', () => {
    render(<Loading dataTestId="testingLoading" isLoading={false} />);

    const loading = screen.queryByTestId('testingLoading');

    expect(loading).not.toBeInTheDocument();
    expect(loading).toMatchSnapshot();
  });
});
