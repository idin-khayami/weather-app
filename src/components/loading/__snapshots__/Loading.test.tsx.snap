// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: loading component should match the snapshot with loading state 1`] = `
<p
  data-testid="testingLoading"
>
  Loading ...
</p>
`;

exports[`Component: loading component should match the snapshot without loading state 1`] = `null`;
